python_modules/libraries/dagster-dg/dagster_dg/cli/component.py

```diff
@@ -65,21 +76,77 @@ def _define_commands(self, cli_context: click.Context) -> None:
             )
             sys.exit(1)
 
-        context = CodeLocationDirectoryContext.from_path(Path.cwd(), app_context)
+        context = CodeLocationDirectoryContext.from_path(Path.cwd(), dg_context)
         for key, component_type in context.iter_component_types():
             command = _create_component_generate_subcommand(key, component_type)
             self.add_command(command)
 
 
-@component_group.group(name="generate", cls=ComponentGenerateGroup)
-def component_generate_group() -> None:
+class ComponentGenerateSubCommand(DgClickCommand):
+    def format_usage(self, context: click.Context, formatter: click.HelpFormatter) -> None:
+        if not isinstance(self, click.Command):
+            raise ValueError("This mixin is only intended for use with click.Command instances.")
+        arg_pieces = self.collect_usage_pieces(context)
+        command_parts = context.command_path.split(" ")
+        command_parts.insert(-1, "[GLOBAL OPTIONS]")
+        return formatter.write_usage(" ".join(command_parts), " ".join(arg_pieces))
+
+    def format_options(self, context: click.Context, formatter: click.HelpFormatter) -> None:
+        # This will not produce any global options since there are none defined on component
+        # generate subcommands.
+        super().format_options(context, formatter)
+
+        # Get the global options off the parent group.
+        parent_context = not_none(context.parent)
```

Comment:
not a big deal, but why not `check.not_none`?

Reply:
`dagster-dg` has no dagster dependency. Didn't want to recreate `check` for one function